src/cmd/usb.js

```diff
@@ -16,6 +16,30 @@ module.exports = class UsbCommand {
 	list(args) {
 		const idsOnly = args['ids-only'];
 		const excludeDfu = args['exclude-dfu'];
+		const filter = args.params.filter;
+
+		let filterFunc = null;
+		if (filter){
+			const platforms = knownPlatforms();
+			if (filter === 'online') {
+				filterFunc = (d) => {
+					return d.connected;
+				};
+			} else if (filter === 'offline') {
+				filterFunc = (d) => {
+					return !d.connected;
+				};
+			} else if (Object.keys(platforms).indexOf(filter) >= 0) {
+				filterFunc = (d) => {
+					return d.platform_id === platforms[filter];
+				};
+			} else {
+				filterFunc = (d) => {
+					return d.id === filter || d.name === filter;
+				};
+			}
+		}
```

Comment:
instead of copy-pasta ([source](https://github.com/particle-iot/particle-cli/blob/2c1cc519fe1b255c88bcbdb61241844249234d30/src/cmd/cloud.js#L497-L518)), let's extract a utility fn :+1: